undefined reference to `jl_cpuid_tag'

Build failure on Raspberry Pi (ARM) at 1c605d2 of master branch.
My previous successful build was at c38a5a3.
```
raspberrypi% make
    LINK usr/bin/julia
repl.o: In function `main':
/home/sachs/src/julia-master/ui/repl.c:248: undefined reference to `jl_cpuid_tag'
collect2: error: ld returned 1 exit status
Makefile:70: recipe for target '/home/sachs/src/julia-master/usr/bin/julia' failed
make[1]: *** [/home/sachs/src/julia-master/usr/bin/julia] Error 1
Makefile:93: recipe for target 'julia-ui-release' failed
make: *** [julia-ui-release] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -96,7 +96,7 @@ julia-inference : julia-base julia-ui-$(JULIA_BUILD_MODE) $(build_prefix)/.examp
 	@$(MAKE) $(QUIET_MAKE) -C $(BUILDROOT) $(build_private_libdir)/inference.ji JULIA_BUILD_MODE=$(JULIA_BUILD_MODE)
 
 ifneq ($(CPUID_SPECIFIC_BINARIES), 0)
-CPUID_TAG = _$(call spawn,$(JULIA_EXECUTABLE) --cpuid)
+CPUID_TAG = _$(call exec,$(JULIA_EXECUTABLE) --cpuid)
 else
 CPUID_TAG =
 endif

--- a/src/Makefile
+++ b/src/Makefile
@@ -103,8 +103,8 @@ DEBUGFLAGS += $(FLAGS)
 SHIPFLAGS += $(FLAGS)
 
 # if not absolute, then relative to the directory of the julia executable
-SHIPFLAGS += "-DJL_SYSTEM_IMAGE_PATH=\"$(build_private_libdir_rel)/sys\""
-DEBUGFLAGS += "-DJL_SYSTEM_IMAGE_PATH=\"$(build_private_libdir_rel)/sys-debug\""
+SHIPFLAGS += "-DJL_SYSTEM_IMAGE_PATH=\"$(build_private_libdir_rel)/sys.$(SHLIB_EXT)\""
+DEBUGFLAGS += "-DJL_SYSTEM_IMAGE_PATH=\"$(build_private_libdir_rel)/sys-debug.$(SHLIB_EXT)\""
 
 ifneq ($(CPUID_SPECIFIC_BINARIES), 0)
 override CPPFLAGS += "-DCPUID_SPECIFIC_BINARIES=1"

--- a/src/jloptions.c
+++ b/src/jloptions.c
@@ -19,15 +19,17 @@ char *shlib_ext = ".dylib";
 char *shlib_ext = ".so";
 #endif
 
-static char default_sysimg_path[512];
+static char system_image_path[256] = "\0" JL_SYSTEM_IMAGE_PATH;
 static const char *get_default_sysimg_path()
 {
-#ifndef CPUID_SPECIFIC_BINARIES
-    snprintf(default_sysimg_path, 512, "%s%s", JL_SYSTEM_IMAGE_PATH, shlib_ext);
-#else
-    snprintf(default_sysimg_path, 512, "%s_%llx%s", JL_SYSTEM_IMAGE_PATH, jl_cpuid_tag(), shlib_ext);
+#ifdef CPUID_SPECIFIC_BINARIES
+    char *path = &system_image_path[1];
+    size_t existing_length = strlen(path) - strlen(shlib_ext);
+    path += existing_length;
+    snprintf(path, sizeof(system_image_path) - existing_length,
+        "_%" PRIx64 "%s", jl_cpuid_tag(), shlib_ext);
 #endif
-    return default_sysimg_path;
+    return &system_image_path[1];
 }
 
 

--- a/src/sys.c
+++ b/src/sys.c
@@ -460,6 +460,12 @@ JL_DLLEXPORT uint64_t jl_cpuid_tag(void)
 }
 #elif defined(CPUID_SPECIFIC_BINARIES)
 #error "CPUID not available on this CPU. Turn off CPUID_SPECIFIC_BINARIES"
+#else
+// For architectures that don't have CPUID
+JL_DLLEXPORT uint64_t jl_cpuid_tag(void)
+{
+    return 0;
+}
 #endif
 
 JL_DLLEXPORT int jl_uses_cpuid_tag()

--- a/ui/repl.c
+++ b/ui/repl.c
@@ -18,6 +18,7 @@
 #include <errno.h>
 #include <math.h>
 #include <ctype.h>
+#include <inttypes.h>
 
 #include "uv.h"
 #define WHOLE_ARCHIVE
@@ -177,7 +178,7 @@ static NOINLINE int true_main(int argc, char *argv[])
     return 0;
 }
 
-extern uint64_t jl_cpuid_tag();
+extern JL_DLLEXPORT uint64_t jl_cpuid_tag();
 
 #ifndef _OS_WINDOWS_
 int main(int argc, char *argv[])
@@ -245,7 +246,7 @@ int wmain(int argc, wchar_t *argv[], wchar_t *envp[])
 #endif
     if (argc >= 2 && strcmp((char *)argv[1], "--cpuid") == 0) {
         /* Used by the build system to name CPUID-specific binaries */
-        printf("%llx", jl_cpuid_tag());
+        printf("%" PRIx64, jl_cpuid_tag());
         return 0;
     }
     libsupport_init();

